applications/MeshingApplication/custom_processes/metrics_hessian_process.cpp

```diff
@@ -77,21 +74,17 @@ void ComputeHessianSolMetricProcess<TDim, TVarType>::Execute()
 {
     // Iterate in the nodes
     NodesArrayType& nodes_array = mThisModelPart.Nodes();
-    int num_nodes = nodes_array.end() - nodes_array.begin();
+    const int num_nodes = nodes_array.end() - nodes_array.begin();
     
     CalculateAuxiliarHessian();
     
     #pragma omp parallel for 
-    for(int i = 0; i < num_nodes; ++i) 
-    {
+    for(int i = 0; i < num_nodes; ++i) {
         auto it_node = nodes_array.begin() + i;
         
-    #ifdef KRATOS_DEBUG 
-        if ( it_node->SolutionStepsDataHas( mVariable ) == false )
-        {
-            KRATOS_ERROR << "Missing variable on node " << it_node->Id() << std::endl;
-        }
-    #endif
+        // Some checks
+        KRATOS_ERROR_IF_NOT(it_node->SolutionStepsDataHas( mVariable )) << "ERROR:: Missing variable on node " << it_node->Id() << std::endl;
```

Comment:
There's a method in the VariablesUtils that performs this kind of check.

Reply:
If you want to check a unique node:
~~~
    const auto& r_node = *mrModelPart.NodesBegin();
    KRATOS_CHECK_VARIABLE_IN_NODAL_DATA(NODAL_H, r_node);
    KRATOS_CHECK_VARIABLE_IN_NODAL_DATA(DISTANCE, r_node);
~~~